reflect_abi_patch.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"maps"
+	"os"
+	"slices"
+	"strings"
+)
+
+func abiNamePatch(path string) (string, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	find := `return unsafe.String(n.DataChecked(1+i, "non-empty string"), l)`
+	replace := `name := unsafe.String(n.DataChecked(1+i, "non-empty string"), l)
+	return _realName(name)`
+
+	str := strings.Replace(string(data), find, replace, 1)
+
+	realname := `
+//go:linkname _realName
+func _realName(name string) string
+`
+
+	return str + realname, nil
+}
+
+var patchMainFile = ""
+
+func reflectMainPrePatch(path string) ([]byte, error) {
+	if patchMainFile != "" {
+		// already patched another file in main
+		return nil, nil
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	patchMainFile = path
+
+	nameMap := "\nvar _nameMap = map[string]string{}"
+
+	return append(content, []byte(realNameCode+nameMap)...), nil
+}
+
+func reflectMainPostPatch(file []byte, lpkg *listedPackage) []byte {
+	obfMapName := hashWithPackage(lpkg, "_nameMap")
+	nameMap := fmt.Sprintf("%s = map[string]string{", obfMapName)
+
+	values := ""
+	obfToReal := realForReflected()
+
+	keys := slices.Collect(maps.Keys(obfToReal))
+	slices.Sort(keys)
+
+	for _, obf := range keys {
+		values += fmt.Sprintf(`"%s": "%s",`, obf, obfToReal[obf])
```

Comment:
use either `strings.Builder` or `text.Template` for this - adding strings will cause a lot of unnecessary allocations and copying of bytes